src/System.Text.Json/src/System/Text/Json/Serialization/JsonClassInfo.cs

```diff
@@ -360,60 +398,52 @@ public JsonPropertyInfo GetProperty(ReadOnlySpan<byte> propertyName, ref ReadSta
 
         public JsonPropertyInfo PolicyProperty { get; private set; }
 
-        private static bool TryIsPropertyRefEqual(in PropertyRef propertyRef, ReadOnlySpan<byte> propertyName, ulong key, ref JsonPropertyInfo info)
-        {
-            if (key == propertyRef.Key)
-            {
-                if (propertyName.Length <= PropertyNameKeyLength ||
-                    // We compare the whole name, although we could skip the first 6 bytes (but it's likely not any faster)
-                    propertyName.SequenceEqual(propertyRef.Info.Name))
-                {
-                    info = propertyRef.Info;
-                    return true;
-                }
-            }
-
-            return false;
-        }
-
-        private static bool IsPropertyRefEqual(ref PropertyRef propertyRef, PropertyRef other)
-        {
-            if (propertyRef.Key == other.Key)
-            {
-                if (propertyRef.Info.Name.Length <= PropertyNameKeyLength ||
-                    propertyRef.Info.Name.AsSpan().SequenceEqual(other.Info.Name.AsSpan()))
-                {
-                    return true;
-                }
-            }
-
-            return false;
-        }
-
+        /// <summary>
+        /// Get a key from the property name.
+        /// If property name length is 7 or less then this key can be used for equality since
+        /// the first 7 bytes are encoded along with the length.
+        /// </summary>
         public static ulong GetKey(ReadOnlySpan<byte> propertyName)
         {
             ulong key;
```

Comment:
If we init key to 0 up front, can we lose the last else branch? Might be worth it.

Reply:
But that would cause an extra initialization for the other cases. (?)

Reply:
I believe it's better to avoid branches, and assignment isn't very expensive.

Also, without the extra branch, we produce smaller assembly:
[sharplab.io - with ints](https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQxNMAfAAgJgIwCwAUFgMwAEu5AwuQN7HlOUUCWAdjOQLIAUwAewEAbcgjhQArsJgBKeo2ZKOXAB4BuRUqasAZuV7ipM2Vu0Mi2q+VXkAvOQCsmy9YC+ZpXGFQ4n5hbW2rYOAAwuQR6uVlgA7DYRzFFKZmTkKjw4/EKiRtJyCtHaGSHk4f46+oYS+aZFSoFBzKXOFeTJ1nEJZlFuQA==)

[sharplab.io - with ulong](https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQxNMAfAAgJgIwCwAUFgMwAEu5AwuQN7HlOUUCuANgPYB2A5uQFkAFME6d25BHCgcYASnqNmyjj34APANxLlTAJYAzckKkz28nboZFdt8uvIBecgFZtNuwF9LyuOyhwPszWdroOzgAM7qHeHrZYAOz20cyxypZk5Kp8gjgiYhKmsgohdtkaTuRRQfpGJtLFNYpxofaVbk1pdonJlrGeQA===)

[sharplab.io - whole method](https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQxNMAfAAgJgIwCwAUFgAwAEWeAdAEoCuAdjAJZhzUCSziA9gA4BlRADcWAYzhQA3MWJYAzJRzkAwuQDexcjspKqANnL0ANr0YBzcgHE4MANJwAngApacCABMA8oxNPBfghGAB5gJ3gAPnJ+BAFEGCcAOQh2AEptXSytIiy83VNzKwBrZ1lc/LyWZnITOEsYAAtyAF4YuP4E5NSOABl6iybyzMrdAHox8gBRMGA4T3ImuHb4hESU9nJqxcblgDMWBFhyAHZycPgocl49neXSp2oR0a3blzqG5uiTjIqXzWe/x0D1a5AAsnAwLwEE4wRAjo0ICY6B5PCFCpZIi5Yqt1j00sM/kCdBM1HV4XdatAYOcIssoLwtjTxMFyI1eCYFktagMmk8icTSQAhZbicxQFiweo08KUChQRosPY01YQGDQq7VKVea63ABki0ZECwABZyPVPCxgowpJqoFB6FJ+cSsiC2iCQiFyKQAB4ADnIkWivr95WJAF9AZU4CYoMsleR3ryvuQFL9iTkXbo3eDIdDYfCFUiUV50dUYFicZ01t10oSszoE0nPq02j8o/9Mw3Xc5yBg2i4MRY0lWuhs4ABtE0AXXInu9PpwpA73b7iaHI46Y56E4ArLP576cH6V92MOuzJZN7ja5ODAevb6FKQww3IwKgTG469Ex9Bs0WjaAx0wbLtV3IEF+wvIpr2rPF2CnB8FyXU8G3PQdL2HUca3HPckKPUNUPyd9uy/eM3j/JpW3IXcQKzMDV0ggcN2w+DJxnOdH0XF8iLyEigX4/4yJ/Zt/0Dcg8Dol4GOJHMIShGE4QRYt3FLegFWhCtsS3HD8XrBsm0ogC2jTXjsjMyomOgq9WNvCccHwn08B4j9/kEl53NGYTDOTajJN4mSgRzWzcNIadXzc3iyICiySUmABBRhzTAfhEnIRgei2K4RCRFhPGdBscxciM5FcvJSRmOYuV2HkW22bkTGpWl4AK/4rIwmC4T5MFqm0m9x2ofpPiQBcADFRrSTiFwUQiiF4rAzgeCK+OeZ5FEoPAjCHGw7EcJwcDcVFfH8QJgjCOlohC/EV0C/JtqK5bRnLWqxLaK72EG5N9P+CrZnmSl3vjJLuQOI4aTOC4pF1SkHlal4fJbb4pNGW6XjkvNFMLRFkVUtEh0rHS2IJUru1JVRyQQSlGuOSHyAZJlyBZJL2U5KnkzhwVJhFRnxUleAallMg6cVZVrmrNUNS2RhtQWG5yANdVyGNM0LStRgbXtLKHSdWKIN7d1e0Pf1A2Df1HsqTzo1jcjf186JTLK8zHcs/XcwUgtlJx1Ey2YAn+r0kmDIo3zANOZGgVR2TeygjqbMJuyOKNlDnazdCWPj3D9ymgjddGNPMNg7cEPvbOfWfc3IpTvJvODltQ+A3XI6C6PmILwHENL5PwPyfOYPbrOk9m7tLaE62RKM6jaMb3Psxb6ysIzndE64pcK483iR+rseEbE6J/Krpv8nR92lKLL21I0tY/bg29ibmqush3qjQ4d7tD7RufY4X/2EIc0vnLXqMTeWRgG6BrrbOubR94Zl1sFReCEwqAN0O+Xiv0qpszqsDGq1MZR0g5i7JwfZW6dTVI0agPVGB9RvgNIa/4Rq+nGpNI2M1vovAWnrJwa93zhiAA===)

It's 1 fewer instruction if you init up front.